exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetFilterPredicate.java

```diff
@@ -18,5 +18,17 @@
 package org.apache.drill.exec.expr.stat;
 
 public interface ParquetFilterPredicate<T extends Comparable<T>> {
-  boolean canDrop(RangeExprEvaluator<T> evaluator);
+
+  /**
+   * Define the validity of a row group against a filter
+   * <ul>
+   *   <li>ALL : all rows match the filter (can not drop the row group and prune filter)
+   *   <li>NONE : no row matches the filter (can drop the row group)
+   *   <li>SOME : some rows only match the filter (can not drop the row group not the filter)
+   *   <li>INAPPLICABLE : filter can not be applied
```

Comment:
Suppose I have set/group of rows and a filter. Applying the filter to the set of rows may result in an empty set (`NONE`), partial set (`SOME`) and the same set as original one (`ALL`). What does `INAPPLICABLE` mean? Does it mean the result is unknown? But does it really matter whether it is partial or unknown?

Reply:
Good question, which we discussed here as well.
The answer is, it does not matter right now. It would matter if we want to stat the filter accurately for example, or if we want to have a sample of results only (faster but inaccurate).
Suppressing INAPPLICABLE is possible, but we'd loose some information of course. keeping it is a plus.

Reply:
Agree with Vlad, I don't see any value of keeping inapplicable.

Reply:
Merged INAPPLICABLE into SOME.
Done